src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Net46.cs

```diff
@@ -0,0 +1,1193 @@
+                // unique connection group name, disposing a handler will remove all these unique connection groups to
+                // save resources.
+                ServicePointManager.CloseConnectionGroups(_connectionGroupName);
+            }
+            base.Dispose(disposing);
+        }
+
+        #endregion De/Constructors
+
+        #region Request Setup
+
+        private HttpWebRequest CreateAndPrepareWebRequest(HttpRequestMessage request)
+        {
+            HttpWebRequest webRequest = null;
+
+            // If we have a request-content, make sure to provide HWR with a delegate to CopyTo(). This allows HWR
+            // to serialize the content multiple times in case of redirect/authentication.
+            // Also note that the connection group name provided is considered an 'internal' connection group. I.e.
+            // HWR will add 'I>' after the string we provided. I.e. by default the actual connection group name looks 
+            // like '123456S>I>' or '123456U>I>' if UnsafeAuthenticatedConnectionSharing is true. Even is users use the
+            // same hashcode for their HWR connection group, they'll end up using a different one, since 'I>' is not
+            // added ('123456S>' or '123456U>').
+            if (request.Content != null)
+            {
+                webRequest = new HttpWebRequest(request.RequestUri, true, _connectionGroupName, request.Content.CopyTo);
+            }
+            else
+            {
+                webRequest = new HttpWebRequest(request.RequestUri, true, _connectionGroupName, null);
+            }
+
+            if (Logging.On) Logging.Associate(Logging.Http, request, webRequest);
+
+            webRequest.Method = request.Method.Method;
+            webRequest.ProtocolVersion = request.Version;
+
+            SetDefaultOptions(webRequest);
+            SetConnectionOptions(webRequest, request);
+            SetServicePointOptions(webRequest, request);
+            SetRequestHeaders(webRequest, request);
+            SetContentHeaders(webRequest, request);
+
+            // New properties for this OOB HttpClientHandler.
+            webRequest.ServicePoint.ConnectionLimit = _maxConnectionsPerServer;
+            webRequest.MaximumResponseHeadersLength = _maxResponseHeadersLength;
+            if ((ClientCertificateOptions == ClientCertificateOption.Manual)
+                && (_clientCertificates != null) && (_clientCertificates.Count > 0))
+            {
+                webRequest.ClientCertificates = _clientCertificates;
+            }
+
+            if (_serverCertificateCustomValidationCallback != null)
+            {
+                webRequest.ServerCertificateValidationCallback = ServerCertificateValidationCallback;
+            }
+
+            if (_defaultProxyCredentials != null && webRequest.Proxy != null)
+            {
+                webRequest.Proxy.Credentials = _defaultProxyCredentials;
```

Comment:
This is a port of the fix from #15036.  I don't plan to change the implementation for the 1.1 port.